File: app/internal/store/document_repo.go
package store

// 数据层：文档(documents)相关查询与维护（均为 *Store 方法，满足 DocumentRepository）。

import "heritage-mgmt/internal/domain"

func (s *Store) ListDocuments(projectID int64) ([]domain.Document, error) {
	rows, err := s.db.Query("SELECT id,project_id,doc_type,doc_type_name,title,orig_name,file_path,file_ext,file_size,uploaded,source FROM documents WHERE project_id=? ORDER BY id", projectID)
	if err != nil {
		return nil, err
	}
	defer rows.Close()
	var out []domain.Document
	for rows.Next() {
		var d domain.Document
		rows.Scan(&d.ID, &d.ProjectID, &d.DocType, &d.DocTypeName, &d.Title, &d.OrigName,
			&d.FilePath, &d.FileExt, &d.FileSize, &d.Uploaded, &d.Source)
		out = append(out, d)
	}
	return out, nil
}

func (s *Store) CountDocs(projectID int64) int {
	var n int
	s.db.QueryRow("SELECT COUNT(*) FROM documents WHERE project_id=?", projectID).Scan(&n)
	return n
}

// DocByID 按ID查文档
func (s *Store) DocByID(id int64) (*domain.Document, error) {
	var d domain.Document
	err := s.db.QueryRow("SELECT id,project_id,doc_type,doc_type_name,title,orig_name,file_path,file_ext,file_size,uploaded,source FROM documents WHERE id=?", id).
		Scan(&d.ID, &d.ProjectID, &d.DocType, &d.DocTypeName, &d.Title, &d.OrigName, &d.FilePath, &d.FileExt, &d.FileSize, &d.Uploaded, &d.Source)
	if err != nil {
		return nil, err
	}
	return &d, nil
}

// DeleteDocument 删除文档记录(不删文件)
func (s *Store) DeleteDocument(id int64) {
	s.db.Exec("DELETE FROM documents WHERE id=?", id)
}

// InsertDocument 插入一条文档记录
func (s *Store) InsertDocument(d domain.Document) error {
	_, err := s.db.Exec(`INSERT INTO documents(project_id,doc_type,doc_type_name,title,orig_name,file_path,file_ext,file_size,uploaded,source) VALUES(?,?,?,?,?,?,?,?,?,?)`,
		d.ProjectID, d.DocType, d.DocTypeName, d.Title, d.OrigName, d.FilePath, d.FileExt, d.FileSize, d.Uploaded, d.Source)
	return err
}

// DeleteDocsByType 删除某工程某分类的全部文档记录，返回删除条数
func (s *Store) DeleteDocsByType(pid int64, docType string) (int64, error) {
	res, err := s.db.Exec("DELETE FROM documents WHERE project_id=? AND doc_type=?", pid, docType)
	if err != nil {
		return 0, err
	}
	return res.RowsAffected()
}

File: desktop/src/core/storage/DocumentRepo.cpp
#include "DocumentRepo.h"

#include <QSqlQuery>

namespace heritage {

DocumentRepo::DocumentRepo(QSqlDatabase db) : db_(db) {}

QVector<Document> DocumentRepo::list(qint64 projectId) {
    QVector<Document> out;
    QSqlQuery q(db_);
    if (!q.prepare(QStringLiteral(
            "SELECT id,project_id,doc_type,doc_type_name,title,orig_name,"
            "file_path,file_ext,file_size,uploaded,source FROM documents WHERE project_id=? ORDER BY id")))
        return out;
    q.addBindValue(projectId);
    if (!q.exec())
        return out;
    while (q.next()) {
        Document d;
        d.id = q.value(0).toLongLong();
        d.projectId = q.value(1).toLongLong();
        d.docType = q.value(2).toString();
        d.docTypeName = q.value(3).toString();
        d.title = q.value(4).toString();
        d.origName = q.value(5).toString();
        d.filePath = q.value(6).toString();
        d.fileExt = q.value(7).toString();
        d.fileSize = q.value(8).toLongLong();
        d.uploaded = q.value(9).toString();
        d.source = q.value(10).toString();
        out.append(d);
    }
    return out;
}

int DocumentRepo::count(qint64 projectId) {
    QSqlQuery q(db_);
    if (!q.prepare(QStringLiteral("SELECT COUNT(*) FROM documents WHERE project_id=?")))
        return 0;
    q.addBindValue(projectId);
    if (!q.exec() || !q.next())
        return 0;
    return q.value(0).toInt();
}

} // namespace heritage
